ICRCs/ICRC-107/ICRC-107.md

```diff
@@ -0,0 +1,206 @@
+
+- If `icrc107_fee_col` is set to an account, all subsequent fees are collected by that account.
+- If `icrc107_fee_col` is not provided (or set to `null`), all subsequent fees are burned.
+
+The `icrc107_set_fee_collection` method MUST return an error in the following cases:
+
+- The caller is not authorized to modify fee collection settings.
+- The provided `Account` is invalid (e.g., malformed principal or subaccount)."
+
+
+### 3.2 `icrc107_get_fee_collection`
+
+This method retrieves the currently active fee collection settings. Unless changed, these settings apply to the next block added to the ledger.
+
+```
+icrc107_get_fee_collection: () -> (opt Account) query;
+```
+
+
+This method should return the currently active fee collection settings:
+
+  - If the response is `null`, fees are burned. This corresponds to `icrc107_fee_col = variant { Array = vec {}}` on-chain.
+  - If the response is a valid `Account`, fees are collected by that account. This corresponds to `icrc107_fee_col` being set to the ICRC-3 representation of the account on-chain.
+
+This method strictly returns the last explicitly set value of `icrc107_fee_col`. It does not infer defaults, and if no fee collector was ever set, it returns `opt Account = null`.
+
+---
+
+
+## 4. Reporting Compliance
+
+Compliance with the current standard is reported as follows.
+
+### 4.1 Supported Standards
+
+Ledgers implementing ICRC-107 **MUST** indicate their compliance through the `icrc1_supported_standards` and `icrc10_supported_standards` methods, by including in the output of these methods:
+
+```
+variant { Vec = vec {
+    record {
+        "name"; variant { Text = "ICRC-107" };
+        "url"; variant { Text = "https://github.com/dfinity/ICRC/blob/main/ICRCs/ICRC-107.md" };
+    }
+}};
+```
+
+### 4.2 Supported Block Types
+
+Ledgers implementing ICRC-107 **MUST** report the new block type in response  to `icrc3_supported_block_types`.  The output of the call must include
+
+```
+variant { Vec = vec {
+    record { "name"; variant { Text = "107feecol" }; "url"; variant { Text = "https://github.com/dfinity/ICRC/blob/main/ICRCs/ICRC-107.md" } };
+}};
+```
+
+## 5. Note on Legacy Fee Collection Mechanisms
+
+The Dfinity maintained ICRC ledgers include a fee collection mechanism which, for completeness is described below.
+
+### 5.1 Legacy Behavior (`fee_col`)
+
+- If `fee_col` is set in a block, the designated account collects only transfer fees (`1xfer`, `2xfer`). Fees for all other operations (e.g., `2approve`) were always burned in legacy behavior as implemented in Dfinity-maintained ICRC-3 ledgers.
+- If `icrc107_fee_col` is not set, the ledger follows this legacy behavior, using `fee_col` only for transfers.
+
+New implementations SHOULD avoid using `fee_col` and instead use `icrc107_fee_col` for all fee collection settings. Legacy behavior is provided for backward compatibility only and MAY be deprecated in future versions of this standard.
```

Comment:
maybe we should provide a context/hint here that existing ledgers can be upgraded and thus might still have to use `fee_col` for backwards compatibility

Reply:
I don't understand what you mean. Can you give suggest some phrasing?